src/hooks/redirection.ts

```diff
@@ -1,31 +1,67 @@
+import { useCallback } from 'react'
 import { useLocation } from 'react-router-dom'
+import { locations } from '../shared/locations'
 
 export const useAfterLoginRedirection = () => {
   const location = useLocation()
   const search = new URLSearchParams(location.search)
+
+  // Extract 'redirectTo' and 'targetConfigId' from current search parameters
   const redirectToSearchParam = search.get('redirectTo')
-  const redirectTo = redirectToSearchParam ? decodeURIComponent(redirectToSearchParam) : null
+  const targetConfigId = search.get('targetConfigId')
+
+  // Initialize redirectTo with a default value
+  let redirectTo = locations.home()
 
-  if (redirectTo === null) {
-    return undefined
+  // Decode 'redirectTo' if it exists
+  if (redirectToSearchParam) {
+    try {
+      redirectTo = decodeURIComponent(redirectToSearchParam)
+    } catch (error) {
+      console.error("Can't decode redirectTo parameter")
+    }
   }
 
+  let sanitizedRedirectTo = locations.home()
+
   try {
-    let redirectToURL: URL
+    let redirectToURL
 
+    // Create a URL object from 'redirectTo'
     if (redirectTo.startsWith('/')) {
       redirectToURL = new URL(redirectTo, window.location.origin)
     } else {
       redirectToURL = new URL(redirectTo)
     }
 
+    // Check if the hostname matches to prevent open redirects
     if (redirectToURL.hostname !== window.location.hostname) {
-      return undefined
+      redirectToURL = new URL('/auth/invalidRedirection', window.location.origin)
+    }
+
+    // Manipulate the search parameters of the redirect URL
+    const redirectToURLSearchParams = new URLSearchParams(redirectToURL.search)
+
+    // Include 'targetConfigId' if it exists
+    if (targetConfigId) {
+      redirectToURLSearchParams.set('targetConfigId', targetConfigId)
```

Comment:
Do we need to add the targetConfigId to the URL to be redirected? This is usually a URL that is outside of the auth site.

Reply:
We are using to redirect to the request page:

https://github.com/decentraland/auth/blob/14be3b569d06e17c390ab4996636efea66c7ee80/src/components/Pages/RequestPage/RequestPage.tsx#L59-L66